### ISSUE TITLE
[lightbox-media-viewer]: CDN link returns 404

### ISSUE THREAD

**Issue by andy-blum** [carbon-design-system/carbon-for-ibm-dotcom#10636]
JIRA: https://jsw.ibm.com/browse/ADCMS-3976

### Description

[This codepen](https://codepen.io/andy-blum/pen/rNQmYvZ) is attempting to load all C4IBM web components from their CDN links. Viewing the network tab, we can see that the dds-lightbox-media-viewer is 404ing.

[lightbox-media-viewer canary docs](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/?path=/docs/components-lightbox-media-viewer--default#lightbox-media-viewer)

[1.33.0 CDN link](https://1.www.s81c.com/common/carbon-for-ibm-dotcom/version/v1.33.0/lightbox-media-viewer.min.js)
[Latest CDN link](https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/latest/lightbox-media-viewer.min.js)

---

Update 9/20/2023: There are three components defined in C4IBM that appear to be unreachable via CDN imports:

- `<dds-lightbox-media-viewer>`
- `<dds-left-nav-item>`
- `<dds-quote-link-with-icon>`

**Comment by kennylam** [comment]
@andy-blum There is no CDN file for `lightbox-media-player` (not sure why, actually). You need to use 

```
https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/latest/lightbox-video-player.min.js
```

Updating your pen with this URI works. You can also see an example of it [here](https://stackblitz.com/github/carbon-design-system/carbon-for-ibm-dotcom/tree/main/packages/web-components/examples/codesandbox/components/lightbox-media-viewer). Check the CDN example directly [here](https://qgnilbjgbgithub-kstn--9000--e809191e.local-credentialless.webcontainer.io/cdn.html).

**Comment by m4olivei** [comment]
> @andy-blum There is no CDN file for lightbox-media-player (not sure why, actually). 

Been digging a little. It looks like the reason this is the case is that the build process normally only bundles (via rollup) from the `index.ts` file within each `packages/web-components/src/components/[component-name]` folder. There are a couple special cases, one is [`lightbox-media-viewer`, where we get picky about which files are the entry points used](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/tools/get-rollup-config.js#L123-L130). `index.ts` is skipped over, leaving out the one we're looking for here, `lightbox-media-viewer`.

For the other two missing from the list @andy-blum put together, `left-nav-item` and `quote-link-with-icon`:

* `left-nav-item` is under `packages/web-components/src/components/masthead`. It's not included statically (from quick glance) from `index.ts` in there, and there is no special handling for the `masthead` folder in the rollup config, so there are no sub-components of `masthead` that are included on their own. We could add that.
* `quote-link-with-icon` is a similiar story.

@andy-blum where did you come up with the list that you came up with? It's not inclusive of all `@customElement` declarations, so wondering what the expectation is here? Seems like there are some components that don't have CDN links, but that we also don't _expect_ to have CDN links?

**Comment by m4olivei** [comment]
Just noticed that in the case of `<dds-lightbox-media-viewer>` its especially obvious b/c the CDN links pointed to in the [Storybook docs](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components/?path=/docs/components-lightbox-media-viewer--default) are broken links:

```
// SPECIFIC VERSION (available starting v1.6.0)
<script type="module" src="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/version/v1.37.0-rc.0/lightbox-media-viewer.min.js"></script>


// LATEST tag
<script type="module" src="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/latest/lightbox-media-viewer.min.js"></script>


// NEXT tag
<script type="module" src="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/tag/v1/next/lightbox-media-viewer.min.js"></script>
```

That feels like an obvious one we'd want to fix. I'll throw up a PR for that.

**Comment by m4olivei** [comment]
> That feels like an obvious one we'd want to fix. I'll throw up a PR for that.

We got this [merged](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11005) to `main`.

> For the other two, @kennylam is there any reason those subcomponents shouldn't be included in their folder's index.ts file?

Was taking a closer look into `<dds-left-nav-item>`. It doesn't look like `<dds-left-nav-item>` is used anymore in masthead v2. That would explain why it's not imported from anywhere that I can see. Appears to be technical debt that needs to be cleaned up. Checked with both Andy and John on this. I've filed a PR to clean this up, against @jkaeser 's [branch to merge masthead v2 with C4IBM v2](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10952). That's been merged to @jkaeser 's branch and will ship with v2. We didn't see value shipping this improvement with v1.

As for `<dds-quote-link-with-icon>`, it looks like that's a special case of `<dds-link-with-icon>` that was introduced later. It looks like the `quote/index.ts` should be updated to import that as well, so that if you do use the CDN link (or ESM import), you have that component available for use like it's demonstrated in the Story. I've done that here https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11020.

Once https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11020 is merged, we can consider this issue finished.
